Assets/Scripts/Entity/Player/PlayerController.cs

```diff
@@ -148,19 +154,31 @@ private void OnRightClick(InputAction.CallbackContext obj)
             if (Physics.Raycast(clickRay, out RaycastHit hit, 10000, clickableLayers) && _attackLogic.Status == AttackLogic.AttackStatus.None)
             {
                 GameObject objectHit = hit.collider.gameObject;
+                Interactable interactable = objectHit.GetComponent<Interactable>();
                 Damageable damageable = objectHit.GetComponent<Damageable>();
                 if (damageable != null)
                 {
                     // implementation NOT capable of area damage
                     _attackLogic.StartAttack(objectHit);
                 }
+
+                else if (interactable != null)
```

Comment:
```suggestion
                else if (objectHit.GetComponent<Interactable>() is Interactable interactable)
```
I'd rewrite this a little so that the variable declaration and the null check are joined together, because we don't need the variables outside of this scope, so this would be a little clearer :) And if you're already on it, can you change the damageable check the same way? 😅

Reply:
Alright !